Core/Src/main.c

```diff
@@ -87,17 +89,41 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);  // Start PWM on channel 1
+  uint8_t rxData;
+  uint8_t txData;
```

Comment:
Refer to Figure 6-1 from the manual. Little hint, it should be an array. You don't transmit/receive only one byte.

Reply:
https://github.com/UWARG/embedded-bootcamp/pull/93/commits/1ae5f8dd0c2503592949c4a3e3f4bc067608070c is this what you mean? I just created a three size buffer and then changed everything to accomodate that basically